src/main/java/net/mcreator/blockly/java/ProcedureCodeOptimizer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.blockly.java;
+
+public class ProcedureCodeOptimizer {
+	/**
+	 * This method attempts to remove the parentheses surrounding the given code, if they are paired.
+	 * @param code The code to optimize
+	 * @return If possible, the code without surrounding parentheses
+	 */
+	public static String removeParentheses(String code) {

```

Comment:
This algorithm will fail if eg. inside the code there is a string containing ( or ), or a comment with ( or )

Reply:
I added a check for strings and comments marked with `/**/`. It doesn't check for comments marked with `//`, but since the only downside is an extra pair of parentheses, it shouldn't cause too many issues

Reply:
Maybe you could use a similar concept as in JavaMemberExtractor for a complete support

Reply:
Other than that, PR is ok, but I would extend this function to handle all syntax cases like JavaMemberExtractor does for extracting member names, while skipping all possible comment and string types

Reply:
I changed the method so that it's more similar to the one in JavaMemberExtractor